packages/web-components/fast-components/src/radio-group/radio-group.ts

```diff
@@ -0,0 +1,171 @@
+    keyCodeArrowLeft,
+    keyCodeArrowRight,
+    keyCodeArrowUp,
+} from "@microsoft/fast-web-utilities";
+import { FASTRadio } from "../radio";
+
+export class RadioGroup extends FASTElement {
+    @attr({ attribute: "readonly", mode: "boolean" })
+    public readOnly: boolean; // Map to proxy element
+    private readOnlyChanged(): void {
+        const filteredRadios = this.getFilteredRadioButtons();
+        if (filteredRadios !== undefined) {
+            filteredRadios.forEach((radio: HTMLElement) => {
+                if (this.disabled) {
+                    radio.setAttribute("readonly", "");
+                } else {
+                    radio.removeAttribute("readonly");
+                }
+            });
+        }
+    }
+
+    @attr({ attribute: "disabled", mode: "boolean" })
+    public disabled: boolean; // Map to proxy element
+    private disabledChanged(): void {
+        const filteredRadios = this.getFilteredRadioButtons();
+        if (filteredRadios !== undefined) {
+            filteredRadios.forEach((radio: HTMLElement) => {
+                if (this.disabled) {
+                    radio.setAttribute("disabled", "");
+                } else {
+                    radio.removeAttribute("disabled");
+                }
+            });
+        }
+    }
+
+    @attr
+    public name: string; // Map to proxy element
+    protected nameChanged(): void {
+        this.getFilteredRadioButtons().forEach((radio: HTMLElement) => {
+            radio.setAttribute("name", this.name);
+        });
+    }
+
+    @attr
+    public value: string;
+    private valueChanged(): void {
+        this.$emit("change");
+    }
+
+    @observable slottedRadioButtons: Node[];
+    private selectedRadio: FASTRadio | HTMLElement;
+
+    constructor() {
+        super();
+        this.addEventListener("keydown", this.keydownHandler);
+    }
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+
+        const radioButtons: HTMLElement[] = this.getFilteredRadioButtons();
+        radioButtons.forEach((radio: HTMLElement) => {
+            radio.addEventListener("change", this.handleRadioChange);
+            if (this.name !== undefined) {
+                radio.setAttribute("name", this.name);
+            }
+
+            if (this.disabled) {
+                radio.setAttribute("disabled", "");
+            }
+
+            if (this.readOnly) {
+                radio.setAttribute("readonly", "");
+            }
+
+            if (this.value && this.value === radio.getAttribute("value")) {
+                this.selectedRadio = radio;
+                radio.setAttribute("checked", "");
+                radio.setAttribute("tabindex", "0");
+            } else {
+                radio.setAttribute("tabindex", "-1");
+            }
+        });
+
+        if (this.value === undefined) {
+            radioButtons[0].setAttribute("tabindex", "0");
+        }
+    }
+
+    private getFilteredRadioButtons = (): any[] => {
```

Comment:
can this be `HTMLElement[]`?

Reply:
it should be FASTRadio[] now and I think better typing, there is still an any in the cast to push into a typed array, but now that's the only any anywhere

Reply:
@marjonlynch I don't know that it will always be FASTRadio. It could be any HTMLElement with a role of radio, yes?

Reply:
ok that is a good point

Reply:
I think that's fine, but I don't know that we can expect it to always be FASTRadio

Reply:
this is fixed as far as the type but I think I might still need to put a check in to see that the item actually conforms to the RadioControl interface. Maybe that interface name needs to change as well, naming is hard

Reply:
confirmed with Nick offline that the current RadioControl interface version satisfies this request.